modules/Components/src/client/ACASThingBrowser.coffee

```diff
@@ -0,0 +1,313 @@
+		for display in @toDisplay
+			@$(".bv_firstRow").append("<th style=\"width: 125px;\">#{display.name}</th>")
+
+		if @collection.models.length is 0
+			@$(".bv_noMatchingThingsFoundMessage").removeClass "hide"
+			# display message indicating no results were found
+		else
+			@$(".bv_noMatchingThingsFoundMessage").addClass "hide"
+			@collection.each (thing) =>
+				prsc = new ACASThingBrowserRowSummaryController
+					model: thing
+					toDisplay: @toDisplay
+				prsc.on "gotClick", @selectedRowChanged
+				@$("tbody").append prsc.render().el
+
+			@$("table").dataTable oLanguage:
+				sSearch: "Filter results: " #rename summary table's search bar
+
+		@
+
+
+class ACASThingBrowserController extends Backbone.View
+	events:
+		"click .bv_deleteThing": "handleDeleteThingClicked"
+		"click .bv_editThing": "handleEditThingClicked"
+		"click .bv_confirmDeleteThingButton": "handleConfirmDeleteThingClicked"
+		"click .bv_cancelDelete": "handleCancelDeleteClicked"
+
+	initialize: (options)->
+		thingModel = new @modelClass
+		@toDisplay = @toDisplay
+		templateVariables = 
+			thingName: thingModel.getName()
+		template = _.template($("#ThingBrowserView").html())
+		$(@el).empty()
+		$(@el).html template(templateVariables)
+		@searchController = new ThingSimpleSearchController
+			model: thingModel
+			query: @query
+			el: @$('.bv_thingSearchController')
+		@searchController.render()
+		@searchController.on "searchReturned", @setupThingSummaryTable.bind(@)
+		@$('.bv_queryToolDisplayName').html window.conf.service.result.viewer.displayName
+
+	setupThingSummaryTable: (things) =>
+		@destroyThingSummaryTable()
+
+		$(".bv_searchingThingsMessage").addClass "hide"
+		if things is null
+			@$(".bv_errorOccurredPerformingSearch").removeClass "hide"
+
+		else if things.length is 0
+			@$(".bv_noMatchingThingsFoundMessage").removeClass "hide"
+			@$(".bv_thingTableController").html ""
+		else
+			@$(".bv_searchThingsStatusIndicator").addClass "hide"
+			@$(".bv_thingTableController").removeClass "hide"
+			thingCollection =  Backbone.Collection.extend 
+				model: @modelClass
+			@thingSummaryTable = new ThingSummaryTableController
+				collection: new thingCollection things
+				toDisplay: @toDisplay
+
+			@thingSummaryTable.on "selectedRowUpdated", @selectedThingUpdated
+			$(".bv_thingTableController").html @thingSummaryTable.render().el
+
+	selectedThingUpdated: (thing) =>
+		@trigger "selectedThingUpdated"
+		@thingController = new @controllerClass
+			model: thing
+			readOnly: true
+
+		@$('.bv_thingController').html @thingController.render().el
+		@$(".bv_thingController").removeClass("hide")
+		@$(".bv_thingControllerContainer").removeClass("hide")
+
+		@$('.bv_editThing').show()
+		if window.conf.thing?.editingRoles?
+			editingRoles = window.conf.thing.editingRoles.split(",")
+			if !UtilityFunctions::testUserHasRole(window.AppLaunchParams.loginUser, editingRoles)
+				@$('.bv_editThing').hide()
+
+        # Not allowing deleting Thing right now
```

Comment:
Don't understand this comment as I do see logic around deleting Things below.

Reply:
The code below was commented out when I started due to an issue but I fixed it and now I'm removing the comment about deleting being disabled.